### DISCUSSION TITLE
How to add a link to search results? When clicked this link would show all the document metadata

### DISCUSSION THREAD

**Discussion by steveh2023** [microsoft-search/pnp-modern-search#3494] [Q&A]
Hi,

Thanks for the previous info! 

I was just wondering if you have any info on how to add a hyperlink to the search results so that when clicked a new tab opens showing the document metadata? 

This screenshot shows the rough idea:

https://ibb.co/K7Rq52K

where the bit i would like to add is in the red rectangle....ie a little icon that when clicked opens a new tab and shows the document metadata - it could be a link to the document in the library or a new page with the info populated - the actual way the UI works has not been specified - only that users want a little button to click to show the metadata....

Is there any info/guidance on this please?

Thanks.

**Comment by wobba** [comment]
You would need to create a link passing over some information, and on the the new tab you need a custom SPFx web part to take the information and somehow show it. Or send an ID and have the other web part load the information. Sort of similar to the List properties web part.

**Comment by kasperbolarsen** [comment]
I wonder if you could make it work by using the option to connect two results web parts,  make a master - detail view?

**Reply by wobba** [reply]
That could work..but won't be a new tab which was the question.

**Reply by steveh2023** [reply]
Hi - thanks for the info - a master detail view would i think be ok - it sounds much simpler to do - i have connected web parts in the past - can you give me an idea how to do this please?

I know i can take the url for dispform.aspx and add the id like this:

/forms/DispForm.aspx?ID=6

this master detail idea - whilst not what i originally asked - would probably be ok - can you let me know a bit more on doing this please?

thanks!

**Reply by wobba** [reply]
See docs at https://microsoft-search.github.io/pnp-modern-search/usage/search-results/connections/#search-results
